docs/usage.md

```diff
@@ -287,3 +288,104 @@ podman build . \
 ```
 
 Details: [write the Containerfile](#write-the-dockerfile-or-containerfile), [build the container](#build-the-container)
+
+
+## Example: cargo
+
+### Prefetch dependencies (cargo)
+
+```shell
+mkdir -p playground/pure-rust
+cd playground/pure-rust
+git clone git@github.com:sharkdp/bat.git --branch=v0.22.1
+cachi2 fetch-deps --source bat '{"type": "cargo"}'
+```
+
+### Generate environment variables (cargo)
+
+#### Alternative crates.io
+Cargo must be configured differently when crates.io will be replaced with an alternative registry to crates.io (or at least similar way - something that the [cargo plugin for nexus](https://github.com/sonatype-nexus-community/nexus-repository-cargo) seems to handle) or from a local path.
+
+For enabling an alternative crates.io registry the only required env is 
+
+```
+CARGO_REGISTRIES_MY_REGISTRY_INDEX=https://my-intranet:8080/git/index
+```
+
+[cargo docs on using an alternate registry](https://doc.rust-lang.org/cargo/reference/registries.html#using-an-alternate-registry)
+
+Unfortunately I'm still clueless on how to run an alternative registry (and this don't seem to align with cachi2 ethos), so I'm skipping this option.
+
+#### Replacing crates.io with a local path
+
+TLDR: not possible through environment variables; will need to inject a config file to a specific location
+
+Unfortunately this method does not work with environment variables (see [this cargo issue](https://github.com/rust-lang/cargo/issues/5416)) and requires a .cargo/config.toml file.
+
+It must be placed somewhere relative to where "cargo install/build" will run following the hierarchical structure below
+
+>Cargo allows local configuration for a particular package as well as global configuration. It looks for configuration files in the current directory and all parent directories. If, for example, Cargo were invoked in `/projects/foo/bar/baz`, then the following configuration files would be probed for and unified in this order:
+>-   `/projects/foo/bar/baz/.cargo/config.toml`
+>-   `/projects/foo/bar/.cargo/config.toml`
+>-   `/projects/foo/.cargo/config.toml`
+>-   `/projects/.cargo/config.toml`
+>-   `/.cargo/config.toml`
+>-   `$CARGO_HOME/config.toml` which defaults to:
+ >   -   Windows: `%USERPROFILE%\.cargo\config.toml`
+>    -   Unix: `$HOME/.cargo/config.toml`
+
+Source: [cargo docs](https://doc.rust-lang.org/cargo/reference/config.html#hierarchical-structure)
+
+This is how I configured mine:
+
+cargo_config.toml
+```
+[source.crates-io]
+replace-with = "local"
+
+[source.local]
+local-registry = "/tmp/cachi2-output"
```

Comment:
I imagine cachi2 has a way to handle paths like this one (given the replacements made on pip requirements).
Can I use something already built to create a file like this one and be flexible on this file?

Reply:
In the RequestOutput returned from `fetch_*_source` functions, you can return [ProjectFiles](https://github.com/containerbuildsystem/cachi2/blob/da6a99a106399a17ed79a881783c24897226e5bd/cachi2/core/models/output.py#L102) which can have an `${output_dir}` placeholder

It might look something like this
```python
ProjectFile(
    abspath=project_dir / ".cargo/config.toml",
    template=dedent(
        """
        ...

        [source.local]
        local-registry = "${output_dir}"
        """
    ),
)
```

Environment variables are always better, but if this is the only option it should work

Reply:
added `.cargo/config.toml` as a project file